crates/network/src/diff_serialization.rs

```diff
@@ -54,3 +75,675 @@ impl WorldDiffDeduplicator {
         diff
     }
 }
+
+/// Explicit tag used for WorldChange enum on the wire
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+#[repr(u8)]
+enum WorldChangeTag {
+    Spawn = 0,
+    Despawn = 1,
+    AddComponents = 2,
+    RemoveComponents = 3,
+    SetComponents = 4,
+}
+
+impl From<&WorldChange> for WorldChangeTag {
+    fn from(change: &WorldChange) -> Self {
+        match change {
+            WorldChange::Spawn(_, _) => WorldChangeTag::Spawn,
+            WorldChange::Despawn(_) => WorldChangeTag::Despawn,
+            WorldChange::AddComponents(_, _) => WorldChangeTag::AddComponents,
+            WorldChange::RemoveComponents(_, _) => WorldChangeTag::RemoveComponents,
+            WorldChange::SetComponents(_, _) => WorldChangeTag::SetComponents,
+        }
+    }
+}
+
+impl<'a> From<&NetworkedWorldChange<'a>> for WorldChangeTag {
+    fn from(change: &NetworkedWorldChange<'a>) -> Self {
+        match change {
+            NetworkedWorldChange::Spawn(_, _) => WorldChangeTag::Spawn,
+            NetworkedWorldChange::Despawn(_) => WorldChangeTag::Despawn,
+            NetworkedWorldChange::AddComponents(_, _) => WorldChangeTag::AddComponents,
+            NetworkedWorldChange::RemoveComponents(_, _) => WorldChangeTag::RemoveComponents,
+            NetworkedWorldChange::SetComponents(_, _) => WorldChangeTag::SetComponents,
+        }
+    }
+}
+
+impl TryFrom<u8> for WorldChangeTag {
```

Comment:
Minor nitpick.


This could be placed immediately after the enum to make it harder to miss to update if you touch the enum. It's current far away state may cause it to be missed when glancing through the code